### ISSUE TITLE
Implement 404 error pages design

### ISSUE THREAD

**Issue by rumanti** [ElementalCode/elemental#170]
^

**Comment by PolyEdge** [comment]
How do you do this in Heroku?

**Comment by Firedrake969** [comment]
Just do a 404.html page in

**Comment by matthewr6** [comment]
http://stackoverflow.com/questions/1569682/how-to-create-a-custom-404-page-for-my-django-apache

(it might be better for you to think in terms of "in Django", not "in Heroku")
(also, unrelated, but you're just styling the ToS, right?  not changing them?)

**Comment by PolyEdge** [comment]
I don't know yet. Rumanti said that she needed help with both because of messy css. We have to discuss what's going to happen. My python client is nearly done, I separated it into classes of GitHub, User, AuthUser, Repository, AuthRepository, Organization, AuthOrganization, File, and AuthFile. I'll document it once it's finished. Before when I said I had it, all you could do was log in and edit files, you couldn't do Pull Requests or anything. My client can't create new branches or merge pull requests, although it can edit files in other repos. Ok??

**Comment by matthewr6** [comment]
What's most needed is to clean up the CSS in the `dev` branch.  I've marked a section that I've started working on that doesn't need cleanup.  The more reusable a class in the CSS is, the better.  You know how to `sass --watch` and have sass installed, right?

You should probably just learn the git CLI commands - might be easier that way ;)
Most of us are either using git or GitHub Desktop, and it's been working out pretty well.

For communication, we mostly use Gitter.
https://gitter.im/Firedrake969

**Comment by PolyEdge** [comment]
Yeah, I noticed a few of the reusable classes like "raleway" --Pretty cool. Don't know sass :frowning: I kind of can push from my python command line :grinning: I got it working. I'm not going to share it until I have full support for repos and orgs-should be done soon. Here's a preview:

``` python
import GitHub
git = GitHub.GitHub('Dylan5797', 'thisisnotmypassword')
print(git.user.get_repositories()[0].name)
'Dylan5797/dylan5797.github.io'
f = AuthFile('Dylan5797/ScratchEdit:master/ScratchEdit.pyw', git)
f.edit('New content')
f.push()
```

pretty cool right! It might not turn out exactly as I put it here though.

**Comment by matthewr6** [comment]
Go ahead and use of t, but try not to distract us :package:

**Comment by matthewr6** [comment]
Use it\* ugh mobile

**Comment by PolyEdge** [comment]
Thanks! I will post it after I get orgs figured. I only have 200 lines but it works :stuck_out_tongue:

**Comment by PolyEdge** [comment]
Maybe we can chat on gitter instead of an issue.

**Comment by matthewr6** [comment]
So your first task is to clean up the SCSS.  If you need help with SCSS feel free to ask.

**Comment by PolyEdge** [comment]
In static folder? Any file in particular you want done first?

**Comment by matthewr6** [comment]
style.scss

**Comment by PolyEdge** [comment]
-  [x] sections
-  [x] uniformity
-  [x] test
-  [x] branch push

**Comment by matthewr6** [comment]
uh
what

**Comment by PolyEdge** [comment]
My checklist.

**Comment by matthewr6** [comment]
For?

Also, just don't touch the `.section` and its nested rules - that's the part I've been working on.

**Comment by ghost** [comment]
why not just use the `<section>` tag in html5??

**Comment by matthewr6** [comment]
because
idk
I haven't gotten used to that super-semantic markup yet.

**Comment by ghost** [comment]
k

**Comment by PolyEdge** [comment]
This has gone a bit inactive. Is the design finished yet?

**Comment by matthewr6** [comment]
@ElementalCode/programmers status?

**Comment by TheInitializer** [comment]
do we have a 404 design?

**Comment by matthewr6** [comment]
idk, find one or make one

**Comment by matthewr6** [comment]
@Rumanti you still doing this?

**Comment by TheInitializer** [comment]
![codelet404](https://cloud.githubusercontent.com/assets/6720845/13730430/6184f31c-e925-11e5-8ed9-23de95275da7.png)
for inspiration, here's the 404 page of a code-sharing site I made a while back

**Comment by PolyEdge** [comment]
http://codepen.io/Dylan5797/pen/GZpyVN

**Comment by matthewr6** [comment]
I like it

**Comment by BookOwl** [comment]
Looks good!

**Comment by TheInitializer** [comment]
some sites have a searchbox on the 404 page, that might be a good idea. Searching is really easy to do in django

**Comment by matthewr6** [comment]
@TheInitializer add that

**Comment by TheInitializer** [comment]
If I have time sure

**Comment by matthewr6** [comment]
@Dylan5797 could you put the HTML file in the static root?

**Comment by PolyEdge** [comment]
Oh my god I have commented like three times and it hasn't come through.

Yes, I'll add it.
I'm not dead, I was just without internet for a while.

**Comment by matthewr6** [comment]
k
